test/test.py

```diff
@@ -177,19 +177,20 @@ def test_delay(self):
         num_requests = 3
 
         def webserver(queue):
-            with http.server.HTTPServer(('', 0), NoLogHandler) as httpd:
-                port = httpd.server_address[1]
-                queue.put(port)
-
-                start = 0
-                for _ in range(num_requests):
-                    httpd.handle_request()
-                    end = time.time()
-                    if end - start < wait_time:
-                        queue.put("too fast")
-                        return
-                    start = end
-                queue.put("ok")
+            httpd = http.server.HTTPServer(('', 0), NoLogHandler)
+            port = httpd.server_address[1]
+            queue.put(port)
+
+            start = 0
+            for _ in range(num_requests):
+                httpd.handle_request()
+                end = time.time()
+                if end - start < wait_time:
+                    queue.put("too fast")
+                    return
```

Comment:
If the test fails because `r2e` fails to wait, then the webserver function will return here (line 190) without properly shutting down the http server, giving the following warning when the tests fail:

```
./usr/lib/python3.6/multiprocessing/process.py:93: ResourceWarning: unclosed <socket.socket fd=7, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('0.0.0.0', 42057)>
```

Maybe wrapping the loop in a `try...finally` would be better if you really want to remove the with block

Reply:
Good catch.

Reply:
Done, see force push diff.